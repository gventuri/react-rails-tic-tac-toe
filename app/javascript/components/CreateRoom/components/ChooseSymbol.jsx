import React from "react";
import PropTypes from "prop-types";
import { Form, Row, Col, Card, Image } from "react-bootstrap";
import classNames from "classnames";

const ChooseSymbol = ({ isSelectedChoice, onChange }) => {
  const CHOOSE_OPTIONS = ["x", "o"];

  return (
    <Form.Group controlId="chooseSymbol">
      <Form.Label>Choose your symbol</Form.Label>
      <Row>
        {CHOOSE_OPTIONS.map((choice) => (
          <Col key={choice}>
            <Card
              className={classNames("card-choice", {
                selected: isSelectedChoice(choice),
              })}
              onClick={() => onChange(choice)}
            >
              <Card.Body className="text-center">
                <Image
                  title={`Option: ${choice}`}
                  src={`/assets/choices/${choice}.png`}
                  width="100"
                  height="100"
                />
              </Card.Body>
            </Card>
          </Col>
        ))}
      </Row>
    </Form.Group>
  );
};

ChooseSymbol.propTypes = {
  isSelectedChoice: PropTypes.func,
  onChange: PropTypes.func,
};

ChooseSymbol.defaultProps = {
  isSelectedChoice: () => false,
  onChange: (choice) => {
    console.log("The value has changed to ", choice);
  },
};

export default ChooseSymbol;
