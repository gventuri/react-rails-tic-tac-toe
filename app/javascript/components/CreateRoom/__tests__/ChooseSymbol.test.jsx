import React from "react";
import { render, cleanup } from "@testing-library/react";
import renderer from "react-test-renderer";
import ChooseSymbol from "../components/ChooseSymbol";

afterEach(cleanup);

it("should render symbol picker", () => {
  render(<ChooseSymbol />);

  // Label
  const chooseSymbolLabel = document.querySelector("label");
  expect(chooseSymbolLabel).toBeInTheDocument();
  expect(chooseSymbolLabel).toHaveTextContent("Choose your symbol");

  // Cards
  const cards = document.querySelectorAll(".card");
  expect(cards.length).toBe(2);
});

it("should have the X choice selected by default", () => {
  render(<ChooseSymbol isSelectedChoice={(choice) => choice === "x"} />);

  const selectedCard = document.querySelector(".selected img");
  expect(selectedCard.getAttribute("title")).toBe("Option: x");
});

it("matches snapshots", () => {
  const chooseSymbol = renderer.create(<ChooseSymbol />).toJSON();
  expect(chooseSymbol).toMatchSnapshot();
});
