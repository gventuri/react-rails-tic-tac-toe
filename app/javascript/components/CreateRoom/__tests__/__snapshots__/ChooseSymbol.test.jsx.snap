// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshots 1`] = `
<div
  className="form-group"
>
  <label
    className="form-label"
    htmlFor="chooseSymbol"
  >
    Choose your symbol
  </label>
  <div
    className="row"
  >
    <div
      className="col"
    >
      <div
        className="card-choice card"
        onClick={[Function]}
      >
        <div
          className="text-center card-body"
        >
          <img
            className=""
            height="100"
            src="/assets/choices/x.png"
            title="Option: x"
            width="100"
          />
        </div>
      </div>
    </div>
    <div
      className="col"
    >
      <div
        className="card-choice card"
        onClick={[Function]}
      >
        <div
          className="text-center card-body"
        >
          <img
            className=""
            height="100"
            src="/assets/choices/o.png"
            title="Option: o"
            width="100"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
